### PR TITLE
feat: update time comparison choices (again)

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

Similar to https://github.com/apache/superset/pull/22458, this PR adds Y5Y time comparison choices to make it faster to compare with pre-pandemic times.

### TESTING INSTRUCTIONS

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-chart-controls/src/sections/advancedAnalytics.tsx [hunk 1]
```diff
@@ -121,6 +121,10 @@ export const advancedAnalyticsControls: ControlPanelSectionConfig = {
             ['2 years ago', t('2 years ago')],
             ['156 weeks ago', t('156 weeks ago')],
             ['3 years ago', t('3 years ago')],
+            ['208 weeks ago', t('208 weeks ago')],
```
Comment (john-bodley) [inline superset-frontend/packages/superset-ui-chart-controls/src/sections/advancedAnalytics.tsx:124]: It seems like the Y4Y comparison was removed in https://github.com/apache/superset/pull/22150, though it's not evident—per the PR description and/or comments—as to why, hence I re-added them.

File: superset-frontend/plugins/legacy-plugin-chart-partition/src/controlPanel.tsx [hunk 2]
```diff
@@ -322,6 +322,10 @@ const config: ControlPanelConfig = {
                 ['2 years', t('2 years')],
                 ['156 weeks', t('156 weeks')],
                 ['3 years', t('3 years')],
+                ['208 weeks ago', t('208 weeks ago')],
```
Comment (michael-s-molina) [inline superset-frontend/plugins/legacy-plugin-chart-partition/src/controlPanel.tsx:325]: @john-bodley Is there are reason to keep the time period in weeks when they match the year? Why not use weeks only when the number of weeks is not a multiple of 52?

Comment (justinpark) [reply superset-frontend/plugins/legacy-plugin-chart-partition/src/controlPanel.tsx:325]: As @michael-s-molina mentioned, the custom time shift option can vary in different scenarios. To ensure consistency, it is advisable to override the custom option by utilizing the chartControlRegistry on the vendor side.

Comment (justinpark) [reply superset-frontend/plugins/legacy-plugin-chart-partition/src/controlPanel.tsx:325]: > john-bodley Is there are reason to keep the time period in weeks when they match the year? Why not use weeks only when the number of weeks is not a multiple of 52?

In order to effectively analyze daily and weekly data and account for seasonal trends, it is essential to incorporate the weekly offset.

FYI:
Certain industries or businesses may experience specific patterns or trends on a weekly basis. For example, in the hospitality industry, weekends may have higher demand compared to weekdays. By using weeks as the time period, analysts can effectively capture these patterns and make more accurate predictions or recommendations.

Comment (michael-s-molina) [reply superset-frontend/plugins/legacy-plugin-chart-partition/src/controlPanel.tsx:325]: Thanks @justinpark for the additional context. @john-bodley I wonder if we should make these configurable somehow.

**Comment by john-bodley** [comment]
Closing in favor of implementing an internal custom override. It seems like Y4Y, Y5Y, etc. comparisons are very much organization dependent.

**Pull request closed without merging**